service/controller/v5/blobclient/blob_client.go

```diff
@@ -76,6 +80,32 @@ func GetBlockBlob(ctx context.Context, blobName string, containerURL *azblob.Con
 	return blobData, nil
 }
 
+func GetBlobURL(ctx context.Context, blobName, containerName, storageAccountName, primaryKey string, containerURL *azblob.ContainerURL) (string, error) {
+
+	sharedKeyCredential, err := azblob.NewSharedKeyCredential(storageAccountName, primaryKey)
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+
+	// Set the desired SAS signature values and sign them
+	//with the shared key credentials to get the SAS query parameters.
+	sasQueryParams, err := azblob.BlobSASSignatureValues{
+		BlobName:      blobName,
+		ContainerName: containerName,
+		ExpiryTime:    time.Now().UTC().Add(blobSASValidityTime * time.Hour),
```

Comment:
Does it mean that if the instance is restarted after an hour it will fail to boot?

Reply:
Nope. It is 180days as `blobSASValidityTime` is 4320 https://github.com/giantswarm/azure-operator/blob/dd4ef8de6d268041803dcc81aa3250b46f0881a3/service/controller/v5/blobclient/blob_client.go#L16